Add CLI tests for activation key delete

oshtaier opened the pull request:
Adding coverage for activation key delete using different
approaches
Closes #1705

JacobCallahan commented:
These methods should be split into separate tests. That way the entire test doesn't fail and our results are more representative. Also, delete by name can be removed, as per our discussion in #2163.

JacobCallahan commented:
Actually, we should perform a better verification that the activation key was deleted.
Checking the result of info should work well.
```result = ActivationKey.info({'id': activation_key_id})```

sthirugn commented:
I agree with @JacobCallahan's comments too.  ACK pending comments.

elyezer commented:
+1 on creating three tests:

1. Delete Activation key using organization name
2. Delete Activation key using organization label
3. Delete Activation key using activation key name

oshtaier commented:
Completely re-implemented all methods that we have for activation key delete functionality. Removed all stubs and added extra coverage that was required in them.

oshtaier commented:
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key
..........
----------------------------------------------------------------------
Ran 10 tests in 219.062s

OK
```

elyezer commented on tests/foreman/cli/test_activationkey.py:
```diff
@@ -438,7 +438,7 @@ def test_negative_create_activation_key_3(self):
         gen_string('html'),
     )
-    def test_positive_delete_activation_key_1(self, name):
+    def test_delete_activation_key_by_name(self, name):
```
:+1:

elyezer commented:
I think that for tests where are being used values like organization name should e used activation key name because just the ID is enough when deleting the activation key (`ActivationKey.delete({'id': activation_key_id})) and it could mask other behaviours.

oshtaier commented:
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key_by_org_name
.
----------------------------------------------------------------------
Ran 1 test in 50.046s

OK

nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key_by_org_label
.
----------------------------------------------------------------------
Ran 1 test in 48.110s

OK
```

oshtaier commented:
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key
..........
----------------------------------------------------------------------
Ran 10 tests in 221.699s

OK
```

elyezer commented:
ACK

Ichimonji10 commented:
Third ACK.
